### PR TITLE
[DA-1451] Report on all participants with no Basics survey submitted.

### PR DESCRIPTION
**Pull request by robabram**

### CODE REVIEW

**Review by j-kanuch — APPROVED** [review]

File: rdr_service/dao/metrics_cache_dao.py [hunk 1]
```diff
@@ -1471,10 +1478,11 @@ def get_metrics_cache_sql(self, hpo_id):
                            ELSE 0
                          END AS Multi_Ancestry,
                          CASE
-                           WHEN (PMI_Skip = 1 AND Number_of_Answer=1) OR UNSET = 1
+                           WHEN (PMI_Skip = 1 AND Number_of_Answer=1) OR (HasBasicsSurvey = 1 AND UNSET = 1)
                              THEN 1
                            ELSE 0
-                         END AS No_Ancestry_Checked
+                         END AS No_Ancestry_Checked,
+                         CASE WHEN HasBasicsSurvey = 0 AND UNSET = 1 THEN 1 ELSE 0 END AS Unset_No_Basics
```
Comment (wangy70) [inline rdr_service/dao/metrics_cache_dao.py]: If `HasBasicsSurvey = 0`, UNSET will always be 1. Maybe we can just use  `HasBasicsSurvey = 0`

File: rdr_service/dao/metrics_cache_dao.py [hunk 2]
```diff
@@ -1517,6 +1527,7 @@ def get_metrics_cache_sql(self, hpo_id):
                                 FROM {temp_table_name} ps
                                 LEFT JOIN participant_race_answers q ON ps.participant_id = q.participant_id
                                 WHERE ps.questionnaire_on_the_basics = 1
+                                    OR (ps.questionnaire_on_the_basics = 0 AND q.code_id is NULL)
```
Comment (wangy70) [inline rdr_service/dao/metrics_cache_dao.py]: if `ps.questionnaire_on_the_basics = 0`, q.code_id will always be NULL.
I think it's the same result if we just remove this where clause.

File: rdr_service/dao/metrics_cache_dao.py [hunk 3]
```diff
@@ -1605,10 +1617,11 @@ def get_metrics_cache_sql(self, hpo_id):
                                  ELSE 0
                                END AS Multi_Ancestry,
                                CASE
-                                 WHEN (PMI_Skip = 1 AND Number_of_Answer=1) OR UNSET = 1
+                                 WHEN (PMI_Skip = 1 AND Number_of_Answer=1) OR (HasBasicsSurvey = 1 AND UNSET = 1)
                                    THEN 1
                                  ELSE 0
-                               END AS No_Ancestry_Checked
+                               END AS No_Ancestry_Checked,
+                               CASE WHEN HasBasicsSurvey = 0 AND UNSET = 1 THEN 1 ELSE 0 END AS Unset_No_Basics
```
Comment (wangy70) [inline rdr_service/dao/metrics_cache_dao.py]: same as above

File: rdr_service/dao/metrics_cache_dao.py [hunk 4]
```diff
@@ -1651,6 +1666,7 @@ def get_metrics_cache_sql(self, hpo_id):
                                       FROM {temp_table_name} ps
                                       LEFT JOIN participant_race_answers q ON ps.participant_id = q.participant_id
                                       WHERE ps.questionnaire_on_the_basics = 1
+                                            OR (ps.questionnaire_on_the_basics = 0 AND q.code_id is NULL)
```
Comment (wangy70) [inline rdr_service/dao/metrics_cache_dao.py]: same as above

File: rdr_service/dao/metrics_cache_dao.py (hunk 1 above)
Comment (robabram) [reply rdr_service/dao/metrics_cache_dao.py]: Good point, I've made all these requested changes in the new commit.

File: rdr_service/dao/metrics_cache_dao.py (hunk 2 above)
Comment (robabram) [reply rdr_service/dao/metrics_cache_dao.py]: Done

File: rdr_service/dao/metrics_cache_dao.py (hunk 3 above)
Comment (robabram) [reply rdr_service/dao/metrics_cache_dao.py]: Done.

File: rdr_service/dao/metrics_cache_dao.py (hunk 4 above)
Comment (robabram) [reply rdr_service/dao/metrics_cache_dao.py]: Done.

**Review by wangy70 — APPROVED** [review]

**Pull request merged**